src/Lib/Managers/ItemsManager.vala

```diff
@@ -52,18 +54,25 @@ public class Akira.Lib.Managers.ItemsManager : Object {
         udpate_default_values ();
 
         Models.CanvasItem? new_item;
+        Models.CanvasArtboard? artboard = null;
+
+        foreach (var _artboard in artboards) {
+            if (_artboard.is_inside (event.x, event.y)) {
+                artboard = _artboard;
+            }
+        }
```

Comment:
Did you try using the `get_item_at (event.x, event.y, true);` method?
Since the `Models.CanvasArtboard` extends the `Goo.CanvasItem` we should inherit that method.
With that method you can avoid looping through artboards, and do something like that:
```
var item = get_item_at (event.x, event.y, true);
if (item != null && item is Models.CanvasArtboard) {
     artboard = item as Models.CanvasArtboard;
}
```

Reply:
Yes, the method `get_item_at` is used by the canvas to determine which item si currently being clicked. 

The problem here is that the `get_item_at` function does make use of the `is_item_at` function, which is overridden and used in the `CanvasArtboard` class to get whether or not the Artboard is selected or hovered, condition that is met only when we are pointing the mouse on the Artboard label, for now, and in the future also on the Artboard's edges. 

The `item` variable, thus, is always null or at least it does represent the item which is _behind_ the item we are trying to create inside the Artboard. 

Looping through artboards seems to be inevitable atm, but we shouldn't be concerned at all, since looping through 20 of them only takes 20-30 microseconds, so we could easily loop though something like 20k Artboard in less than a blink of an eye :)